include/deal.II/base/config.h.in

```diff
@@ -277,40 +277,42 @@
 
 #ifdef DEAL_II_COMPILER_HAS_DIAGNOSTIC_PRAGMA
 
-#  define DEAL_II_DISABLE_EXTRA_DIAGNOSTICS                      \
-_Pragma("GCC diagnostic push")                                   \
-_Pragma("GCC diagnostic ignored \"-Wunknown-pragmas\"")          \
-_Pragma("GCC diagnostic ignored \"-Wpragmas\"")                  \
-_Pragma("GCC diagnostic ignored \"-Wunknown-warning-option\"")   \
-_Pragma("GCC diagnostic ignored \"-Wunknown-warning\"")          \
-_Pragma("GCC diagnostic ignored \"-Wextra\"")                    \
-_Pragma("GCC diagnostic ignored \"-Woverloaded-virtual\"")       \
-_Pragma("GCC diagnostic ignored \"-Wunused-function\"")          \
-_Pragma("GCC diagnostic ignored \"-Wunused-parameter\"")         \
-_Pragma("GCC diagnostic ignored \"-Wunused-variable\"")          \
-_Pragma("GCC diagnostic ignored \"-Wtype-limits\"")              \
+/* keep pragmas with an exclamation mark in order.. */
+#  define DEAL_II_DISABLE_EXTRA_DIAGNOSTICS                             \
+_Pragma("GCC diagnostic push")                                    /*!*/ \
+_Pragma("GCC diagnostic ignored \"-Wunknown-pragmas\"")           /*!*/ \
+_Pragma("GCC diagnostic ignored \"-Wpragmas\"")                   /*!*/ \
+_Pragma("GCC diagnostic ignored \"-Wunknown-warning-option\"")    /*!*/ \
+_Pragma("GCC diagnostic ignored \"-Wunknown-warning\"")           /*!*/ \
+_Pragma("GCC diagnostic ignored \"-Wextra\"")                     /*!*/ \
+_Pragma("GCC diagnostic ignored \"-Waddress-of-packed-member\"")        \
+_Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"")         \
+_Pragma("GCC diagnostic ignored \"-Wexpansion-to-defined\"")            \
+_Pragma("GCC diagnostic ignored \"-Wexpansion-to-defined\"")            \
+_Pragma("GCC diagnostic ignored \"-Wignored-attributes\"")              \
+_Pragma("GCC diagnostic ignored \"-Wignored-qualifiers\"")              \
+_Pragma("GCC diagnostic ignored \"-Wimplicit-fallthrough\"")            \
+_Pragma("GCC diagnostic ignored \"-Winfinite-recursion\"")              \
+_Pragma("GCC diagnostic ignored \"-Wint-in-bool-context\"")             \
+_Pragma("GCC diagnostic ignored \"-Wmisleading-indentation\"")          \
+_Pragma("GCC diagnostic ignored \"-Wmissing-field-initializers\"")      \
+_Pragma("GCC diagnostic ignored \"-Wnested-anon-types\"")               \
+_Pragma("GCC diagnostic ignored \"-Wnon-virtual-dtor\"")                \
+_Pragma("GCC diagnostic ignored \"-Woverflow\"")                        \
+_Pragma("GCC diagnostic ignored \"-Woverloaded-virtual\"")              \
+_Pragma("GCC diagnostic ignored \"-Wpedantic\"")                        \
```

Comment:
This is very pedantic but would you mind aligning the `\`s? If you don't I can write a follow-up patch. The wobbly back-slashes bother me more than they should.

Reply:
We're all type-A personalities, @drwells! :-)